FinSearchUnified/Tests/PluginTest.php

```diff
@@ -90,6 +91,97 @@ public function testArticleExport($articlesActiveStatus, $expectedCount, $errorM
         $this->assertEquals($expectedCount, $actual, sprintf($errorMessage, $actual));
     }
 
+    /**
+     * Data provider for the export test cases with corresponding assertion message
+     *
+     * @return array
+     */
+    public function articleProviderWithId()
+    {
+        return [
+            'No article for ID' => [
+                'productId' => 20,
+                'response' => json_encode([
+                    'errors' => [
+                        'general' => ['No article found with ID 20'],
+                        'products' => []
+                    ]
+                ])
+            ],
+            '1 Article found by ID match' => [
+                'productId' => 3,
+                'response' => 1
+            ],
+            '2 Article found by ID and supplier match' => [
+                'productId' => 2,
+                'response' => 2
+            ],
+            '1 Article with error by ID match' => [
+                'productId' => 4,
+                'response' => json_encode([
+                    'errors' => [
+                        'general' => [],
+                        'products' => [
+                            [
+                                'id' => 4,
+                                'errors' => [
+                                    'Product is not active.',
+                                    'Main Detail is not active or not available.',
+                                    'shouldBeExported is false.'
+                                ]
+                            ]
+                        ]
+                    ]
+                ])
+            ],
+            '1 Article without and 2 articles with error' => [
+                'productId' => 1,
+                'response' => json_encode([
```

Comment:
The data provider returns a `response`, but the test expects an `expectedCount`. The parameters should match.